cmd/restic/fuse/dir.go

```diff
@@ -0,0 +1,100 @@
+package fuse
+
+import (
+	"encoding/binary"
+	"os"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+	"golang.org/x/net/context"
+
+	"github.com/restic/restic"
+	"github.com/restic/restic/repository"
+)
+
+// Statically ensure that *dir implement those interface
+var _ = fs.HandleReadDirAller(&dir{})
+var _ = fs.NodeStringLookuper(&dir{})
+
+type dir struct {
+	repo     *repository.Repository
+	children map[string]*restic.Node
+	inode    uint64
+}
+
+func newDir(repo *repository.Repository, node *restic.Node) (*dir, error) {
+	tree, err := restic.LoadTree(repo, node.Subtree)
+	if err != nil {
+		return nil, err
+	}
+	children := make(map[string]*restic.Node)
+	for _, child := range tree.Nodes {
+		children[child.Name] = child
+	}
+
+	return &dir{
+		repo:     repo,
+		children: children,
+		inode:    node.Inode,
+	}, nil
+}
+
+func newDirFromSnapshot(repo *repository.Repository, snapshot SnapshotWithId) (*dir, error) {
+	tree, err := restic.LoadTree(repo, snapshot.Tree)
+	if err != nil {
+		return nil, err
+	}
+	children := make(map[string]*restic.Node)
+	for _, node := range tree.Nodes {
+		children[node.Name] = node
+	}
+
+	return &dir{
+		repo:     repo,
+		children: children,
+		inode:    binary.BigEndian.Uint64(snapshot.ID),
```

Comment:
That's a mistake, it should also be [:8].

Also, inode is a `uint64` while snapshot.ID is a `backend.ID` which is a `[]byte` in disguise. It can't be used as a uint64.